### ISSUE TITLE
[Question] How can I use only light and light italic by default in vscode?

### ISSUE THREAD

**Issue by WindSoilder** [microsoft/cascadia-code#500]
I have tried to build my self based on main branch, and I just allow `CascadiaMono-Roman_Light` and `CascadiaMono-Italic_Light-Italic` font in my font setting.
![图片](https://user-images.githubusercontent.com/22256154/120952975-b156e200-c77e-11eb-9a2a-8e7711f031fb.png)

![图片](https://user-images.githubusercontent.com/22256154/120953000-bddb3a80-c77e-11eb-89fd-5c8a9fb7a43d.png)

But I have issue to make use of them in vscode, all fonts in vscode is italic..
![图片](https://user-images.githubusercontent.com/22256154/120953177-0c88d480-c77f-11eb-8d43-e13447bf1380.png)

If I enable `CascadiaMono-Roman` and `CascadiaMono-Italic`, this issue disappeared... But the weight is too high for me, so I just want to use a lighter version..

**Comment by aaronbell** [comment]
Ok, two things. 

One is that you should use the PR that I submitted last week as that has necessary improvements to the name table (it should be pushed to main this week).

Two, it appears you’re trying to use the Variable font version of the fonts. So either you will need to set the font weight in VSCode to a lighter weight (say, 300), or switch to the stat instances (which the name table fix should help with).

**Comment by WindSoilder** [comment]
Hi, thanks for your quickly reply :-)
But sorry it seems this doesn't working for me.

> One is that you should use the PR that I submitted last week as that has necessary improvements to the name table (it should be pushed to main this week)

I have installed from this source. (I believe the code lays here: https://github.com/aaronbell/cascadia-code/tree/work)

> Two, it appears you’re trying to use the Variable font version of the fonts. So either you will need to set the font weight in VSCode to a lighter weight (say, 300), or switch to the stat instances (which the name table fix should help with).

Here is the screenshort when no setting lighter weight:
![图片](https://user-images.githubusercontent.com/22256154/121104165-03a60a80-c834-11eb-99f2-2ddbc1545108.png)

Here you can see that I'm using the source code from your PR (the italic font shape has changed.)

Here is another screenshort when setting lighter weight (everything goes into italic again):
![图片](https://user-images.githubusercontent.com/22256154/121104251-36500300-c834-11eb-8af1-e8492f5c476e.png)

**Comment by aaronbell** [comment]
Can you zip up all of the fonts that you’re generating And attach them here so I can take a look at what might be going on?

As for `salt` / `ss01`, these are code functions in the font. See https://github.com/microsoft/vscode/issues/80577 for how to implement these in VS Code.

**Comment by WindSoilder** [comment]
Sure, here is the file.
[ttf.tar.gz](https://github.com/microsoft/cascadia-code/files/6612940/ttf.tar.gz)

**Comment by aaronbell** [comment]
Hi, sorry for the delay in getting back to you. Finally had a chance to take a look at the files you sent over and, honestly, I don't see anything wrong with them. 

I suspect that the issue might end up being something related to font conflicts in your font cache. I'd suggest removing all Cascadia Code fonts from your system, restarting, then reinstalling. Also, potentially [clear the font cache](https://github.com/microsoft/cascadia-code/wiki/Double-installation-issue). Sorry for the trouble!

**Comment by spaceaardvark** [comment]
I am seeing similar behavior with the latest release. If I set `editor.fontWeight` to `"300"`, then everything is in italics. Is that what your PR is supposed to fix, @aaronbell? And thank you for your efforts!

**Comment by WindSoilder** [comment]
@spaceaardvark Hi, would you please try to use the ttf I upload from here can solve your issue?

https://github.com/microsoft/cascadia-code/issues/500#issuecomment-856376217

I have tried it again @aaronbell , it seems that it works on my windows machine, but failed in Mac machine, I will try to clear font cache in mac(If there is a cache) when I got it.

**Comment by spaceaardvark** [comment]
@WindSoilder that doesn't fix the problem.

**Comment by aaronbell** [comment]
Hi! Please try the new release and let me know if it resolves your issues. Thanks!

https://github.com/microsoft/cascadia-code/releases/tag/v2106.17

**Comment by WindSoilder** [comment]
Yeah, I'm using Mac, I have tried both `otf` and `ttf`, it seems that `otf` is ok, but `ttf` doesn't.

**Comment by aaronbell** [comment]
Out of interest, can you provide a bit more information about the system / VS Code version that you're using? I feel like this is likely a platform-specific issue but want to make sure.

**Comment by WindSoilder** [comment]
System version: `macOS Big Sur11.4`

Vscode about information:
```
Version: 1.58.1
Commit: 2aeda6b18e13c4f4f9edf6667158a6b8d408874b
Date: 2021-07-13T06:22:33.013Z (20 hrs ago)
Electron: 12.0.13
Chrome: 89.0.4389.128
Node.js: 14.16.0
V8: 8.9.255.25-electron.0
OS: Darwin x64 20.5.0
```

**Comment by aaronbell** [comment]
hmmmm. For fun, what happens if you set the font weight to 301.

**Comment by aaronbell** [comment]
Another thing I'd be interested for you to test out. Can you visit this site: https://vscode-web-test-playground.azurewebsites.net/ (ideally in Chrome) 

Click the triple lines in the upper left `File > Preferences > Settings` and then enter in the code settings:
```
{
    "editor.fontFamily": "\"Cascadia Code\", Menlo, Monaco, 'Courier New', monospace",
    "editor.fontWeight": 300,
    "editor.tokenColorCustomizations": {
        "comments": {"fontStyle": "italic"}
    },
}
```

Does the issue repro there as well?

**Comment by WindSoilder** [comment]
> hmmmm. For fun, what happens if you set the font weight to 301.

Every is still italic.

> Does the issue repro there as well?

Yeah, I can reproduct it. (On chrome)

![图片](https://user-images.githubusercontent.com/22256154/125569987-e98ef7d9-c3d3-413a-abc7-a2791f70e354.png)

**Comment by aaronbell** [comment]
Good news bad news time!

Good news: This issue is isolated to Big Sur, it seems
Bad news: It is definitely something in the font because JetBrains Mono does not have the same problem
Good news: I have identified what is causing the issue and can fix it
Bad news: There appears to be a *secondary* issue that JetBrains Mono also suffers from wherein any weight value less than 400 becomes the lowest possible value (100 for JetBrains, 200 for Cascadia). 

I will need to investigate further tomorrow.

**Comment by s1eke** [comment]
Same problem on win11
![image](https://user-images.githubusercontent.com/10690839/125742346-d40a4c9d-34d3-43d3-8f1a-98f54f41c987.png)

System Build：22000.65
VSCode Version：1.58.1
Cascadia Version：2106.17

**Comment by WindSoilder** [comment]
Thanks !.! I have checked the new build ttf in your pr, and it works!

**Comment by WindSoilder** [comment]
Oh, by the way , I found that `SemiLight` weight value in your pr is `350`, but I have tried to set weight to `350` in vscode, but vscode is still using `Regular` weight rather than `SemiLight`, I don't know if it's the vscode's issue.

**Comment by aaronbell** [comment]
See https://github.com/microsoft/vscode/issues/128097

:/

**Comment by avarayr** [comment]
After some experimentation, I've found that In Vscode font-weight anything from 250 to 349 defaults to all-italic, including the whole number weight 300

![image](https://user-images.githubusercontent.com/7735415/129500329-5392d1d7-f102-4922-aedc-16ad620710f8.png)
